port/unix/omrsock.c

```diff
@@ -31,10 +31,143 @@
 
 #include "omrcfg.h"
 #if defined(OMR_PORT_SOCKET_SUPPORT)
+#include "omrsock.h"
 #include "omrport.h"
 #include "omrporterror.h"
 #include "omrsockptb.h"
 
+/* Internal: OMRSOCK user interface constants TO OS dependent constants mapping. */
+
+/**
+ * @internal Map OMRSOCK API user interface address family names to the OS address 
+ * family, which may be defined differently depending on operating system. 
+ *
+ * @param OMRAddrFamily The OMR address family to be converted.
+ *
+ * @return OS address family, or OS_SOCK_AF_UNSPEC if none exists.
+ */
+static int32_t
+map_addr_family_OMR_to_OS(int32_t OMRAddrFamily)
+{
+	switch(OMRAddrFamily)
+	{
+		case OMRSOCK_AF_INET:
+			return OS_SOCK_AF_INET;
+		case OMRSOCK_AF_INET6:
+			return OS_SOCK_AF_INET6;
+	}
+	return OS_SOCK_AF_UNSPEC;
+}
+
+/**
+ * @internal Map OMRSOCK API user interface socket type to the OS socket type, 
+ * which may be defined differently depending on operating system. 
+ *
+ * @param OMRSockType The OMR socket type to be converted.
+ *
+ * @return OS socket type on success, or OS_SOCK_ANY if none exists.
+ */
+static int32_t
+map_sockettype_OMR_to_OS(int32_t OMRSockType)
```

Comment:
I think calling this `socktype` would be better than `sockettype`, because `socktype` is more consistent with the rest of the code.